lib/TPP/TileConsumerAndFuseProducers.cpp

```diff
@@ -140,10 +272,30 @@ struct TileConsumerAndFuseProducers
     linalg::registerTilingInterfaceExternalModels(registry);
   }
   void runOnOperation() override {
+    func::FuncOp func = getOperation();
+    transform::TrivialPatternRewriter rewriter(&getContext());
+    llvm::SmallDenseSet<Operation *> fusedOps;
+    func->walk([&](linalg::LinalgOp linalgOp) {
+      if (linalg::isElementwise(linalgOp) && hasMatmulLikeProducer(linalgOp)) {
+        SmallVector<int64_t> actualTileSizes = llvm::to_vector(this->tileSizes);
+        // Default tiling scheme, always legal to tile by 1.
```

Comment:
I believe in this case it is a good default. If you have a matmul follow by an element-wise I think it is good to fuse and tile along the parallel dimensions. `tileSizes` comes from the pass option, only if no options are passed we use the default {1, 1}, otherwise we use the tile sizes provided by the user.

Reply:
If we don't like a default I can remove it.

Reply:
No, we want a default, I just think that `{1, 1}` isn't a good default. Tiling a matmul for each element isn't a good idea for almost any hardware.

According to libxsmm experiments, the meaningful default tile sizes are 32 and 64. IIUC, less than this, we don't fill the cache lines, more than that we start clashing lines due to associativity.

In that sense, the default value should be the minimum we recommend doing (in libxsmm, so 32), and only tile lower than that if the pass option is set.

In your testing, you can pass the option `{1, 1}` to the larger test (all alternatives) and add a small test with the default (and check that it's 32).